### PR TITLE
Adding Subscription Language

### PR DESCRIPTION
**Pull request by prasannapolina**
#6978 Add LanguageId field to NewsletterSubscription

1.Added new language field "Subscription Language" in the NewsLetterSubscription table.
2.To store the data in the new field for both guest and subscribed user made changes in the src/Presentation/Nop.Web/Controllers/NewsletterController.cs file

### CODE REVIEW

**Comment by apurvai007** [comment]
**Question** : To complete this request, we need to alter the table `NewsSubscription` and add a new column `SubscriptionLanguage` in `schemaMigration` , correct?

**Comment by skoshelev** [comment]
Hi @prasannapolina. 
@apurvai007 is right, just adding a field to the entity is not enough as that will only work for a clean install. We still need to add it to the migration. Also in the note to the ticket, it is written that it is necessary to take into account the user’s ability to change the language while using the site. That is, the new field should change after Customer.LanguageId.

File: src/Libraries/Nop.Core/Domain/Messages/NewsLetterSubscription.cs [hunk 1]
```diff
@@ -19,6 +19,8 @@ public partial class NewsLetterSubscription : BaseEntity
     /// Gets or sets a value indicating whether subscription is active
     /// </summary>
     public bool Active { get; set; }
+    public int SubscriptionLanguage { get; set; }
```
Comment (skoshelev) [inline src/Libraries/Nop.Core/Domain/Messages/NewsLetterSubscription.cs:22]: Better to use the LanguageId as a name. And also we need a doc string to the new field.

File: src/Presentation/Nop.Web/Controllers/NewsletterController.cs [hunk 2]
```diff
@@ -61,10 +61,14 @@ public virtual async Task<IActionResult> SubscribeNewsletter(string email, bool
         {
             email = email.Trim();
             var store = await _storeContext.GetCurrentStoreAsync();
-            var subscription = await _newsLetterSubscriptionService.GetNewsLetterSubscriptionByEmailAndStoreIdAsync(email, store.Id);
```
Comment (skoshelev) [inline src/Presentation/Nop.Web/Controllers/NewsletterController.cs:64]: I think that changes in these three lines are not necessary

File: src/Presentation/Nop.Web/Controllers/NewsletterController.cs [hunk 3]
```diff
@@ -61,10 +61,14 @@ public virtual async Task<IActionResult> SubscribeNewsletter(string email, bool
...
             email = email.Trim();
             var store = await _storeContext.GetCurrentStoreAsync();
-            var subscription = await _newsLetterSubscriptionService.GetNewsLetterSubscriptionByEmailAndStoreIdAsync(email, store.Id);
             var currentLanguage = await _workContext.GetWorkingLanguageAsync();
+            var subscription = await _newsLetterSubscriptionService.GetNewsLetterSubscriptionByEmailAndStoreIdAsync(email, store.Id);
+            //subscription.SubscriptionLanguage = currentLanguage.Id;
             if (subscription != null)
+                
             {
+                subscription.SubscriptionLanguage = currentLanguage.Id;
```
Comment (skoshelev) [inline src/Presentation/Nop.Web/Controllers/NewsletterController.cs:70]: These changes are not needed since there is no entity saving in this block.

**Comment by DmitriyKulagin** [comment]
Hi @prasannapolina 
We decided to implement this task on our own, since solving it requires several more changes than are proposed here. Still, thank you for your help.

**Pull request closed without merging**